<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jerelle Jesse" />


<title>Trawl Biodiversity Metrics for Landings Comparison</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Seagrant: ME-NH inshore trawl survey</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="biodiversity.html">Biodiversity Metrics</a>
</li>
<li>
  <a href="gamms.html">GAMMs</a>
</li>
<li>
  <a href="Trawl_plots.html">Functional Groups</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    NMDS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="NMDS_species_biomass_7.21.html">Top 50 species biomass</a>
    </li>
    <li>
      <a href="NMDS_species_abundance_7.21.html">Top 50 species abundance</a>
    </li>
    <li>
      <a href="NMDS_group_biomass_7.21.html">Functional group biomass</a>
    </li>
    <li>
      <a href="NMDS_group_abundance_7.21.html">Functional group abundance</a>
    </li>
  </ul>
</li>
<li>
  <a href="anosim_adonis_analysis_7.21.html">Anosim/ Adonis</a>
</li>
<li>
  <a href="env_NMDS.html">Environmental Effects on NMDS</a>
</li>
<li>
  <a href="fishery_landings.html">Fishery Landings</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>




</div>


<style type="text/css">/********** GMRI Rmarkdown Core Style Sheet - Do Not Modify!!! **********/


/********** Begin Style Sheet **********/

/* Avenir Font from Fonts.com for GMRI Branding */

@import url("http://fast.fonts.net/t/1.css?apiType=css&projectid=806f61f6-d695-4965-a878-820b50bc0269");
    @font-face{
        font-family:"Avenir LT W01_35 Light1475496";
        src:url("Fonts/0078f486-8e52-42c0-ad81-3c8d3d43f48e.woff2") format("woff2"),url("Fonts/908c4810-64db-4b46-bb8e-823eb41f68c0.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_35 Light_1475502";
        src:url("Fonts/a59168c1-917d-4de9-a244-0316c057c357.woff2") format("woff2"),url("Fonts/6dc0e7d8-9284-44e1-8f05-984a41daa3a4.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_45 Book1475508";
        src:url("Fonts/065a6b14-b2cc-446e-9428-271c570df0d9.woff2") format("woff2"),url("Fonts/65d75eb0-2601-4da5-a9a4-9ee67a470a59.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_45 Book O1475514";
        src:url("Fonts/476612d9-282d-4f76-95cd-b4dd31e7ed21.woff2") format("woff2"),url("Fonts/f1ebae2b-5296-4244-8771-5f40e60a564a.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_55 Roman1475520";
        src:url("Fonts/b290e775-e0f9-4980-914b-a4c32a5e3e36.woff2") format("woff2"),url("Fonts/4b978f72-bb48-46c3-909a-2a8cd2f8819c.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_55 Obliqu1475526";
        src:url("Fonts/1a7173fa-062b-49ad-9915-bc57d3bfc1f5.woff2") format("woff2"),url("Fonts/cdda031e-26e9-4269-83d1-5a218caa10db.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_65 Medium1475532";
        src:url("Fonts/17b90ef5-b63f-457b-a981-503bb7afe3c0.woff2") format("woff2"),url("Fonts/c9aeeabd-dd65-491d-b4be-3e0db9ae47a0.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_65 Medium1475538";
        src:url("Fonts/deb5e718-7abb-4df3-9365-edfa95317090.woff2") format("woff2"),url("Fonts/04801919-17ee-4c6b-8b17-eb1965cb3ed6.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_85 Heavy1475544";
        src:url("Fonts/d513e15e-8f35-4129-ad05-481815e52625.woff2") format("woff2"),url("Fonts/61bd362e-7162-46bd-b67e-28f366c4afbe.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_85 Heavy_1475550";
        src:url("Fonts/3c210c80-960f-4684-850b-25390b4d08af.woff2") format("woff2"),url("Fonts/cb5c71ad-e582-4d00-929c-67fbfaeb1c27.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_95 Black1475556";
        src:url("Fonts/c78eb7af-a1c8-4892-974b-52379646fef4.woff2") format("woff2"),url("Fonts/75b36c58-2a02-4057-a537-09af0832ae46.woff") format("woff");
    }
    @font-face{
        font-family:"Avenir LT W01_95 Black_1475562";
        src:url("Fonts/a2477e08-09d9-4d4b-97a9-23a1e22cb44c.woff2") format("woff2"),url("Fonts/19d12bba-92b1-43ad-9bab-cd36a4195c2a.woff") format("woff");
    }





/* PRE-Avenir Fonts: Lato + Raleway font import from google fonts */
@import url('https://fonts.googleapis.com/css?family=Lato');
@import url('https://fonts.googleapis.com/css?family=Raleway&display=swap');

/* add font families as needed: font-family: 'Lato', sans-serif; */


/* Level 1 Headers */
h1 { text-align: left;
     margin: 10px 0 15px 0; 
     font-size: 38px; 
     font-family: Avenir;
} 


/* Headers 2 - 6 */
h2, h3, h4, h5, h6 { 
    color: #333333; 
    margin: 20px 0 5px 0; 
    text-align: left;
    font-family: Avenir;}


/* Sizing/font For Each Header Type */
h2 { font-size: 24px; }
h3 { font-size: 20px; }
h4 { font-size: 18px; }
h5 { font-size: 16px; font-weight: normal; color: #3069aa; text-decoration: underline;}
h6 { font-size: 14px; font-weight: normal; color: #3069aa; }


/* Paragraph Text */
p, ol { margin-top: 10px; 
    font-family: 'Raleway', sans-serif;}


/* Title Author and Date Headers */
h1.title.toc-ignore {margin-top: 10px;}
h4.author, h4.date {
    color: rgb(0,115,109);
    margin-top: 0px; 
    margin-bottom: 5px; 
    font-size: 12px;}



/* Add Spacing Above Headers */
h1, .h1, h2, .h2, h3, .h3 {
    margin-top: 40px;
}


/* Links */
a {
    color: rgb(234,79,18)
}


/***********************************************/


/********  Table of Contents  **********/

/* Highlighted TOC Element */
.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: rgb(0,96,138);
    border-color: rgb(0,96,138);
}

/* Default TOC Elements */
.list-group-item, .list-group-item:focus, .list-group-item:hover {
    z-index: 2;
    color: rgb(0,96,138);
    background-color: #fff;
    border-color: rgb(0,96,138);
}


/********  Tab Panels  **********/

/* Navigation Tabs - Highlighted Tabset Pills */
.nav-pills > li.active > a, .nav-pills > li.active > a:hover, .nav-pills > li.active > a:focus {
    color: #fff;
    background-color: rgb(0,115,109) ;
    }

/* Navigation Tabs - Default Tabset Pills */
.nav-pills > li > a, .nav-pills > li > a:hover, .nav-pills > li > a:focus {
    color: rgb(0,115,109);
    background-color: #fff;
    }


/* Second Level Tabs - Active */
.nav.nav-tabs > li.active  > a, .nav-tabs > li.active > a:hover, .nav-tabs > li.active > a:focus {
    color: #fff;
    background-color: rgb(83,83,83) ;
    }

/* Second Level Tabs - Inactive */
.nav.nav-tabs > li  > a, .nav-tabs > li > a:hover, .nav-tabs > li > a:focus {
    color: rgb(83,83,83);
    background-color: #fff;
    }



/********** End Core Style Sheet **********/
/********** End Core Style Sheet **********/



</style>
<div id="me-nh-trawl-diveristy-metrics" class="section level2">
<h2>ME-NH trawl diveristy metrics</h2>
<p>By season and total</p>
<pre class="r"><code>library(here)
library(ggplot2)
library(dplyr)
new_data&lt;-read.csv(here(&quot;Data/MaineDMR_Trawl_Survey_Catch_Data_2021-05-14.csv&quot;))

#remove unnecessary species by getting scientific name from old data
setwd(&quot;C:/Users/jjesse/Box/Kerr Lab/Fisheries Science Lab/ME NH Trawl- Seagrant/Seagrant-AEW/ME NH data for GMRI&quot;)

trawl &lt;- read.csv(&quot;EXPCATCH_forGMRI.csv&quot;, header = TRUE)%&gt;%
  select(COMMON_NAME, SCIENTIFIC_NAME)%&gt;%
  distinct()

trawl_2&lt;- left_join(trawl,new_data)

# first drop samples that are not to the species level
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Anemonia&quot; ,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Pandalus&quot; ,]    #ID to spp also
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Stelleroidea&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Octopoda&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Clypeasteroida&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Yoldia&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Calcarea&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Majidae&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Balanus&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Stomatopoda&quot;,]    #low
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Euphausiacea&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Paguroidea&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Mysidacea&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Diaphus&quot;,]        #low
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Sepiolidae&quot;,]
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Artediellus&quot;,]    #low
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Macrouridae&quot;,]    #low
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Paralepididae&quot;,]  #low
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Clupeidae&quot;,]      #low
trawl_2 &lt;- trawl_2[!trawl_2$SCIENTIFIC_NAME == &quot;Myctophidae&quot;,]    #low

## there are 1121 occurrences where catch = NA; remove these?
trawl_2 &lt;- trawl_2[!is.na(trawl_2$Expanded_Catch),]
## for some reason there 26 are observations of catch = 0; remove these 
trawl_2 &lt;- trawl_2[!trawl_2$Expanded_Catch == 0,]

## there are 16 occurrences where weight = NA; remove these?
trawl_2 &lt;- trawl_2[!is.na(trawl_2$Expanded_Weight_kg),]
## for some reason there 0 are observations of weight = 0; remove these 
trawl_2 &lt;- trawl_2[!trawl_2$Expanded_Catch == 0,]

data&lt;-trawl_2</code></pre>
<div id="species-richness" class="section level3">
<h3>Species richness</h3>
<pre class="r"><code>richness&lt;-group_by(data, Year,Season)%&gt;%
  summarise(richness=length(unique(COMMON_NAME)))

ggplot()+ geom_line(data=richness, aes(Year, richness))+facet_grid(rows = vars(Season))+theme_bw()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<pre class="r"><code>richness&lt;-group_by(data, Year)%&gt;%
  summarise(richness=length(unique(COMMON_NAME)))

rich&lt;-ggplot()+ geom_line(data=subset(richness,Year&gt;2005 &amp; Year&lt;2020), aes(Year, richness))+ theme_classic()+ ylab(expression(paste(&quot;Species\nRichness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

rich</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-3-2.png" width="672" /></p>
</div>
<div id="shannon-weiner-diversity" class="section level3">
<h3>Shannon-Weiner diversity</h3>
<pre class="r"><code>shannon&lt;-group_by(data, Year,Season,COMMON_NAME)%&gt;%
  mutate(species_total=sum(Expanded_Weight_kg, na.rm=TRUE))%&gt;%
  group_by(Year, Season)%&gt;%
  mutate(total=sum(species_total), prop=(species_total/total))%&gt;%
  summarise(shannon=-1*(sum(prop*log(prop),na.rm=TRUE)))

ggplot()+geom_line(data=shannon, aes(Year, shannon))+ 
  facet_grid(rows=vars(Season))+theme_bw()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>shannon&lt;-group_by(data, Year,COMMON_NAME)%&gt;%
  mutate(species_total=sum(Expanded_Weight_kg, na.rm=TRUE))%&gt;%
  group_by(Year)%&gt;%
  mutate(total=sum(species_total),prop=(species_total/total))%&gt;%
  summarise(shannon=-1*(sum(prop*log(prop),na.rm=TRUE)))

diversity&lt;-ggplot()+geom_line(data=subset(shannon,Year&gt;2005 &amp; Year &lt;2020), aes(Year, shannon))+theme_classic()+ ylab(expression(paste(&quot;Shannon-Weiner\nDiversity&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

diversity</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>
</div>
<div id="simpsons-diversity-and-evenness" class="section level3">
<h3>Simpson’s diversity and evenness</h3>
<pre class="r"><code>simpson&lt;-group_by(data, Year, Season, COMMON_NAME)%&gt;%
  mutate(species_total=sum(Expanded_Weight_kg, na.rm=TRUE))%&gt;%
  group_by(Year, Season)%&gt;%
  mutate(richness=length(unique(COMMON_NAME)))%&gt;%
  mutate(total=sum(species_total))%&gt;%
  mutate(prop=(species_total/total))%&gt;%
  summarise(simpsonD=1/(sum(prop^2)), simpsonE=simpsonD*(1/richness))
# this code is not working properly, very high simpson&#39;s d and e??

ggplot()+geom_line(data=simpson, aes(Year, simpsonE))+
  facet_grid(rows=vars(Season))+theme_bw()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre class="r"><code>simpson&lt;-group_by(data, Year,COMMON_NAME)%&gt;%
  summarise(species_total=sum(Expanded_Weight_kg, na.rm=TRUE))%&gt;% #aggregate to get yearly species totals
  group_by(Year)%&gt;%
  mutate(richness=length(unique(COMMON_NAME)))%&gt;%
  mutate(total=sum(species_total))%&gt;%
  mutate(prop=(species_total/total))%&gt;%
  summarise(simpsonD=1/(sum(prop^2)),simpsonE=simpsonD*(1/richness))
   
ggplot()+geom_line(data=simpson, aes(Year, simpsonD))+theme_classic()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-5-2.png" width="672" /></p>
<pre class="r"><code>evenness&lt;-ggplot()+geom_line(data=subset(simpson,Year&gt;2005 &amp; Year &lt;2020), aes(Year, simpsonE))+theme_classic()+ ylab(expression(paste(&quot;Simpson&#39;s\nEvenness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

evenness</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-5-3.png" width="672" /></p>
</div>
<div id="average-taxinomic-distinctness" class="section level3">
<h3>Average taxinomic distinctness</h3>
<pre class="r"><code>library(taxize)
library(purrr)
library(mgcv)

# data2&lt;-read.csv(here(&quot;Data/common_scientific_convert.csv&quot;))%&gt;%
#   distinct(COMMON_NAME, .keep_all=TRUE)
# 
# data2&lt;-left_join(data,data2, by=&quot;COMMON_NAME&quot;)

species &lt;- filter(data,!is.na(SCIENTIFIC_NAME))%&gt;%
  rename(Species=SCIENTIFIC_NAME)
  #add scientific name for taxonomic distinctness

#na&lt;-filter(data, is.na(SCIENTIFIC_NAME))%&gt;%
#  distinct(COMMON_NAME)

diff_species&lt;-as.vector(unique(species$Species))
tax &lt;- classification(diff_species, db = &#39;itis&#39;) </code></pre>
<pre><code>## ==  142 queries  =============
## v  Found:  Alosa pseudoharengus
## v  Found:  Squalus acanthias
## v  Found:  Glyptocephalus cynoglossus
## v  Found:  Paralichthys oblongus
## v  Found:  Pseudopleuronectes americanus
## v  Found:  Limanda ferruginea
## v  Found:  Melanogrammus aeglefinus
## v  Found:  Urophycis chuss
## v  Found:  Merluccius bilinearis
## v  Found:  Urophycis tenuis
## v  Found:  Homarus americanus
## v  Found:  Lophius americanus
## v  Found:  Hippoglossoides platessoides
## v  Found:  Pollachius virens
## v  Found:  Zoarces americanus
## v  Found:  Myoxocephalus octodecemspinosus
## v  Found:  Centropristis striata
## v  Found:  Leucoraja erinacea
## v  Found:  Leucoraja ocellata
## v  Found:  Loligo pealeii
## v  Found:  Gadus morhua
## v  Found:  Cancer irroratus
## v  Found:  Enchelyopus cimbrius
## v  Found:  Placopecten magellanicus
## v  Found:  Prionotus carolinus
## v  Found:  Aspidophoroides monopterygius
## v  Found:  Cancer borealis
## v  Found:  Trachurus lathami
## v  Found:  Hemitripterus americanus
## v  Found:  Illex illecebrosus
## v  Found:  Peprilus triacanthus
## v  Found:  Tautogolabrus adspersus
## v  Found:  Clupea harengus
## v  Found:  Stenotomus chrysops
## v  Found:  Amblyraja radiata
## v  Found:  Sebastes fasciatus
## v  Found:  Scomber scombrus
## v  Found:  Pandalus borealis
## v  Found:  Alosa sapidissima
## v  Found:  Dichelopandalus leptocerus
## v  Found:  Brevoortia tyrannus
## v  Found:  Osmerus mordax
## v  Found:  Cryptacanthodes maculatus
## v  Found:  Menidia menidia
## v  Found:  Ammodytes americanus
## v  Found:  Acipenser oxyrinchus
## v  Found:  Cyclopterus lumpus
## v  Found:  Pandalus montagui
## v  Found:  Arctica islandica
## v  Found:  Caudina arenata
## v  Found:  Crangon septemspinosa
## v  Found:  Engraulis eurystole
## v  Found:  Chionoecetes opilio
## v  Found:  Cucumaria frondosa
## v  Found:  Urophycis regia
## v  Found:  Myoxocephalus scorpius
## v  Found:  Astarte undata
## v  Found:  Hyas araneus
## v  Found:  Lebbeus groenlandicus
## v  Found:  Triglops murrayi
## v  Found:  Citharichthys arctifrons
## v  Found:  Malacoraja senta
## v  Found:  Pitar morrhuanus
## v  Found:  Hippoglossus hippoglossus
## v  Found:  Carcinus maenas
## v  Found:  Cyclocardia borealis
## v  Found:  Chlamys islandica
## v  Found:  Strongylocentrotus droebachiensis
## v  Found:  Modiolus modiolus
## v  Found:  Zenopsis conchifera
## v  Found:  Decapterus punctatus
## v  Found:  Asterias vulgaris
## v  Found:  Maurolicus muelleri
## v  Found:  Torpedo nobiliana
## v  Found:  Lumpenus lumpretaeformis
## v  Found:  Lumpenus maculatus
## v  Found:  Gasterosteus aculeatus
## v  Found:  Reinhardtius hippoglossoides
## v  Found:  Anguilla rostrata
## v  Found:  Selene setapinnis
## v  Found:  Alosa aestivalis
## v  Found:  Syngnathus fuscus
## v  Found:  Pomatomus saltatrix
## v  Found:  Pristigenys alta
## v  Found:  Mytilus edulis
## v  Found:  Lithodes maja
## v  Found:  Peristedion miniatum
## v  Found:  Ulvaria subbifurcata
## v  Found:  Nemichthys scolopaceus
## v  Found:  Chaceon quinquedens
## v  Found:  Mercenaria mercenaria
## v  Found:  Pasiphaea multidentata
## v  Found:  Liparis coheni
## v  Found:  Dipturus laevis
## v  Found:  Lunatia heros
## v  Found:  Selar crumenophthalmus
## v  Found:  Liparis atlanticus
## v  Found:  Myxine glutinosa
## v  Found:  Lebbeus polaris
## v  Found:  Eualus fabricii
## v  Found:  Pandalus propinquus
## v  Found:  Liparis inquilinus
## v  Found:  Liparis liparis
## v  Found:  Neptunea decemcostata
## v  Found:  Anarhichas lupus
## v  Found:  Mallotus villosus
## v  Found:  Microgadus tomcod
## v  Found:  Pontophilus norvegicus
## v  Found:  Morone saxatilis
## v  Found:  Buccinum undatum
## v  Found:  Anchoa hepsetus
## v  Found:  Axius serratus
## v  Found:  Pholis gunnellus
## v  Found:  Raja eglanteria
## v  Found:  Decapterus macarellus
## v  Found:  Scomberesox saurus
## v  Found:  Lycenchelys verrilli
## v  Found:  Myoxocephalus aenaeus
## v  Found:  Liparis fabricii
## v  Found:  Monacanthus hispidus
## v  Found:  Spirontocaris liljeborgii
## v  Found:  Sphoeroides maculatus
## v  Found:  Melanostigma atlanticum
## v  Found:  Polymixia lowei
## v  Found:  Xenolepidichthys dalgleishi
## v  Found:  Spirontocaris spinus
## v  Found:  Colus stimpsoni
## v  Found:  Liparis tunicatus
## v  Found:  Phycis chesteri
## v  Found:  Helicolenus dactylopterus
## v  Found:  Ariomma bondi
## v  Found:  Lepophidium profundorum
## v  Found:  Lophius gastrophysus
## v  Found:  Ostrea edulis
## v  Found:  Menticirrhus saxatilis
## v  Found:  Antigonia combatia
## v  Found:  Brosme brosme
## v  Found:  Apeltes quadracus
## v  Found:  Sclerocrangon boreas
## v  Found:  Eumicrotremus spinosus
## v  Found:  Leucoraja garmani
## v  Found:  Ensis directus
## ==  Results  =================
## 
## * Total: 142 
## * Found: 142 
## * Not Found: 0</code></pre>
<pre class="r"><code>info &lt;- matrix(NA)
expand &lt;- matrix(NA)
specific &lt;- matrix(NA, nrow = length(diff_species), ncol=6)


for (i in 1:length(tax)){
info &lt;- tax[[i]][c(&#39;name&#39;,&#39;rank&#39;)]
expand &lt;- info[info$rank == &#39;phylum&#39;| info$rank == &#39;class&#39;| info$rank == &#39;order&#39; | info$rank == &#39;family&#39; | info$rank == &#39;genus&#39; | info$rank == &#39;species&#39;,]
specific[i,] &lt;- as.vector(expand$name)

}
colnames(specific) &lt;- c(&quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;, &quot;Species&quot;)
phylo&lt;-as.data.frame(specific)

merge&lt;-left_join(species,phylo, by=&quot;Species&quot;)
  
data_tax&lt;-merge%&gt;%
  filter(!is.na(Species))

data_tax_groups&lt;-group_by(data_tax, Year,Species)%&gt;%
  summarise(catch=sum(Expanded_Weight_kg, na.rm=TRUE))%&gt;%
  mutate(indicator=cur_group_id())%&gt;%
  left_join(phylo)

hauls &lt;- unique(data_tax_groups$indicator)   
N_hauls &lt;- length(hauls) # number of hauls
N_species &lt;- NULL #N species
sub_species &lt;- NULL # N_species-1
total &lt;- NULL  #xixj
numerator &lt;- NULL #wijxixj
x_y &lt;- matrix(NA, nrow = 6, ncol = 6) 
x &lt;- NULL
y &lt;- NULL
ident &lt;- NULL
weight &lt;- NULL #wij
count &lt;- NULL
total_weight &lt;- NULL
mean_weight &lt;- NULL
weight_var &lt;- NULL
delta &lt;- NULL
delta_star &lt;- NULL
delta_plus &lt;- NULL
delta_var &lt;- NULL
weight_var &lt;- NULL

for (j in 1:N_hauls) {  
diff_hauls &lt;- data_tax_groups[which(data_tax_groups$indicator == j),] #subset unique hauls/functional groups
N_species[j] &lt;- length(unique(diff_hauls$Species))# count the number of unique species in each haul (denominator)
sub_species[j] &lt;- N_species[j]-1
diff &lt;- unique(as.vector(diff_hauls$Species)) # name of each unique species 
combos &lt;- combn(diff, 2) # create combinations of each species/haul (for weight calc)

phylo &lt;- as.matrix(subset(diff_hauls, select = c(Phylum,Class,Order,Family,Genus, Species))) # extract phylogenetic information only
unique_phylo &lt;- uniquecombs(phylo) # subset by unique species information
unique_phylo &lt;- as.data.frame(unique_phylo)

total &lt;- NULL  # reset the length for each haul because they will be different
weight &lt;- NULL # reset  

for (i in 1:ncol(combos)) { # for each unique combination count the number of each species 
  total[i] &lt;- sum(diff_hauls$Species == combos[1,i]) * sum(diff_hauls$Species == combos[2,i]) #empty vector is always length 210
  #total[i] &lt;- diff_hauls[diff_hauls[,22] == combos[1,i],9] * diff_hauls[diff_hauls[,22] == combos[2,i],9]
  x &lt;- unique_phylo[unique_phylo$Species == combos[1,i],]
  y &lt;- unique_phylo[unique_phylo$Species == combos[2,i],]
  x_y &lt;- rbind(x,y)
  
  for (k in 1:ncol(x_y)){ # for each combination calculate the weight value 
    ident[k] &lt;- identical(as.vector(x_y[1,k]), as.vector(x_y[2,k])) # determine how much of phylogenetic information is the same
    weight[i] &lt;- sum(ident == &quot;FALSE&quot;) # vector of weights
    #mean_weight[i] &lt;- mean(weight) #rep(mean(weight),length(weight))
    numerator[j] &lt;- sum(total*weight) 
    count[j] &lt;- sum(total)
    mean_weight[j] &lt;- mean(weight)
    total_weight[j] &lt;- sum(weight)
    weight_var[j] &lt;- sum((weight- mean(weight))^2) 
  }
  delta &lt;- (2*numerator)/(N_species*sub_species)
  delta_star &lt;- numerator/(count)
  delta_plus &lt;- (2*total_weight)/(N_species*sub_species)
  delta_var &lt;- (2*weight_var)/(N_species*sub_species) #double check that this equation is correct
}
}
years&lt;-2006:2020
delta&lt;-as.data.frame(delta[1:15]) # taxonomic diversity
delta_star&lt;-as.data.frame(delta_star[1:15]) # taxonomic distinctness
delta_plus&lt;-as.data.frame(delta_plus[1:15]) # average taxonomic distinctness
delta_var&lt;-as.data.frame(delta_var[1:15]) # variation in taxonomic distinctness

d&lt;-bind_cols(years,delta, delta_star,delta_plus,delta_var)
colnames(d)&lt;-c(&quot;Year&quot;, &quot;delta&quot;, &quot;delta_star&quot;, &quot;delta_plus&quot;,&quot;delta_var&quot;)

#write.csv(d, here(&quot;Data/tax_metrics.csv&quot;))

tax.distinct&lt;-ggplot()+geom_line(data=subset(d,Year&gt;2005 &amp; Year &lt;2020), aes(Year, delta_plus))+theme_classic()+ylab(expression(paste(&quot;Average\nTaxinomic\nDistinctness&quot;)))+theme(plot.margin = margin(10, 10, 10, 25))
                                                                                     
tax.distinct</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>Combo plot</p>
<pre class="r"><code>library(reshape2)
library(gridExtra)

all_metrics&lt;-full_join(richness,shannon, by=&quot;Year&quot;)%&gt;%
  full_join(simpson)%&gt;%
  full_join(d)

#write.csv(all_metrics, here(&quot;Data/trawl_biodiversity_update.csv&quot;))

all_melt&lt;-melt(all_metrics, id=&quot;Year&quot;)

ggplot()+geom_line(data=all_melt, aes(Year,value))+facet_grid(rows = vars(variable), scales=&quot;free&quot;)+theme_bw()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre class="r"><code>grid.arrange(rich, diversity, evenness,tax.distinct, nrow=4)</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-7-2.png" width="672" /></p>
</div>
</div>
<div id="regional-diversity" class="section level2">
<h2>Regional diversity</h2>
<pre class="r"><code>data_regions&lt;-data
# group to east/west of Penobscot Bay
data_regions$group&lt;-NULL
data_regions$group[data_regions$Region==1]&lt;-&quot;East&quot;
data_regions$group[data_regions$Region==2]&lt;-&quot;East&quot;
data_regions$group[data_regions$Region==3]&lt;-&quot;Penobscot&quot;
data_regions$group[data_regions$Region==4]&lt;-&quot;West&quot;
data_regions$group[data_regions$Region==5]&lt;-&quot;West&quot;

data_regions&lt;-filter(data_regions,!is.na(group))%&gt;%
  filter(Expanded_Weight_kg!=0)</code></pre>
<div id="species-richness-1" class="section level3">
<h3>Species richness</h3>
<pre class="r"><code>richness&lt;-group_by(data_regions, group, Year)%&gt;%
  summarise(richness=length(unique(SCIENTIFIC_NAME)))

ggplot()+ geom_line(data=richness, aes(Year, richness))+
  facet_grid(rows=vars(group))+theme_bw()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
<div id="shannon-weiner-diversity-1" class="section level3">
<h3>Shannon-Weiner diversity</h3>
<pre class="r"><code>shannon&lt;-group_by(data_regions, Year,group)%&gt;%
  mutate(total=sum(Expanded_Weight_kg, na.rm=TRUE), prop=(Expanded_Weight_kg/total))%&gt;%
  summarise(shannon=-1*(sum(prop*log(prop))))

ggplot()+geom_line(data=shannon, aes(Year, shannon))+
  facet_grid(rows=vars(group))+theme_bw()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
<div id="simpsons-diversity-and-evenness-1" class="section level3">
<h3>Simpson’s diversity and evenness</h3>
<pre class="r"><code>simpson&lt;-group_by(data_regions, Year,group,SCIENTIFIC_NAME)%&gt;%
  summarise(species_total=sum(Expanded_Weight_kg, na.rm=TRUE))%&gt;% #aggregate 
  group_by(Year, group)%&gt;%
  mutate(richness=length(unique(SCIENTIFIC_NAME)))%&gt;%
  mutate(total=sum(species_total))%&gt;%
  mutate(prop=species_total/total)%&gt;%
  summarise(simpsonD=1/(sum(prop^2)),simpsonE=simpsonD*(1/richness))

ggplot()+geom_line(data=simpson, aes(Year, simpsonE))+
  facet_grid(rows=vars(group))+theme_bw()</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
<div id="average-taxinomic-distinctness-1" class="section level3">
<h3>Average taxinomic distinctness</h3>
<pre class="r"><code>library(taxize)
library(purrr)
library(mgcv)
library(gridExtra)

species &lt;- filter(data_regions,!is.na(SCIENTIFIC_NAME))%&gt;%
  rename(Species=SCIENTIFIC_NAME)
  

diff_species&lt;-as.vector(unique(species$Species))
tax &lt;- classification(diff_species, db = &#39;itis&#39;) </code></pre>
<pre><code>## ==  142 queries  =============
## v  Found:  Alosa pseudoharengus
## v  Found:  Squalus acanthias
## v  Found:  Glyptocephalus cynoglossus
## v  Found:  Paralichthys oblongus
## v  Found:  Pseudopleuronectes americanus
## v  Found:  Limanda ferruginea
## v  Found:  Melanogrammus aeglefinus
## v  Found:  Urophycis chuss
## v  Found:  Merluccius bilinearis
## v  Found:  Urophycis tenuis
## v  Found:  Homarus americanus
## v  Found:  Lophius americanus
## v  Found:  Hippoglossoides platessoides
## v  Found:  Pollachius virens
## v  Found:  Zoarces americanus
## v  Found:  Myoxocephalus octodecemspinosus
## v  Found:  Centropristis striata
## v  Found:  Leucoraja erinacea
## v  Found:  Leucoraja ocellata
## v  Found:  Loligo pealeii
## v  Found:  Gadus morhua
## v  Found:  Cancer irroratus
## v  Found:  Enchelyopus cimbrius
## v  Found:  Placopecten magellanicus
## v  Found:  Prionotus carolinus
## v  Found:  Aspidophoroides monopterygius
## v  Found:  Cancer borealis
## v  Found:  Trachurus lathami
## v  Found:  Hemitripterus americanus
## v  Found:  Illex illecebrosus
## v  Found:  Peprilus triacanthus
## v  Found:  Tautogolabrus adspersus
## v  Found:  Clupea harengus
## v  Found:  Stenotomus chrysops
## v  Found:  Amblyraja radiata
## v  Found:  Sebastes fasciatus
## v  Found:  Scomber scombrus
## v  Found:  Pandalus borealis
## v  Found:  Alosa sapidissima
## v  Found:  Dichelopandalus leptocerus
## v  Found:  Brevoortia tyrannus
## v  Found:  Osmerus mordax
## v  Found:  Cryptacanthodes maculatus
## v  Found:  Menidia menidia
## v  Found:  Ammodytes americanus
## v  Found:  Acipenser oxyrinchus
## v  Found:  Cyclopterus lumpus
## v  Found:  Pandalus montagui
## v  Found:  Arctica islandica
## v  Found:  Caudina arenata
## v  Found:  Crangon septemspinosa
## v  Found:  Engraulis eurystole
## v  Found:  Chionoecetes opilio
## v  Found:  Cucumaria frondosa
## v  Found:  Urophycis regia
## v  Found:  Myoxocephalus scorpius
## v  Found:  Astarte undata
## v  Found:  Hyas araneus
## v  Found:  Lebbeus groenlandicus
## v  Found:  Triglops murrayi
## v  Found:  Citharichthys arctifrons
## v  Found:  Malacoraja senta
## v  Found:  Pitar morrhuanus
## v  Found:  Hippoglossus hippoglossus
## v  Found:  Carcinus maenas
## v  Found:  Cyclocardia borealis
## v  Found:  Chlamys islandica
## v  Found:  Strongylocentrotus droebachiensis
## v  Found:  Modiolus modiolus
## v  Found:  Zenopsis conchifera
## v  Found:  Decapterus punctatus
## v  Found:  Asterias vulgaris
## v  Found:  Maurolicus muelleri
## v  Found:  Torpedo nobiliana
## v  Found:  Lumpenus lumpretaeformis
## v  Found:  Lumpenus maculatus
## v  Found:  Gasterosteus aculeatus
## v  Found:  Reinhardtius hippoglossoides
## v  Found:  Anguilla rostrata
## v  Found:  Selene setapinnis
## v  Found:  Alosa aestivalis
## v  Found:  Syngnathus fuscus
## v  Found:  Pomatomus saltatrix
## v  Found:  Pristigenys alta
## v  Found:  Mytilus edulis
## v  Found:  Lithodes maja
## v  Found:  Peristedion miniatum
## v  Found:  Ulvaria subbifurcata
## v  Found:  Nemichthys scolopaceus
## v  Found:  Chaceon quinquedens
## v  Found:  Mercenaria mercenaria
## v  Found:  Pasiphaea multidentata
## v  Found:  Liparis coheni
## v  Found:  Dipturus laevis
## v  Found:  Lunatia heros
## v  Found:  Selar crumenophthalmus
## v  Found:  Liparis atlanticus
## v  Found:  Myxine glutinosa
## v  Found:  Lebbeus polaris
## v  Found:  Eualus fabricii
## v  Found:  Pandalus propinquus
## v  Found:  Liparis inquilinus
## v  Found:  Liparis liparis
## v  Found:  Neptunea decemcostata
## v  Found:  Anarhichas lupus
## v  Found:  Mallotus villosus
## v  Found:  Microgadus tomcod
## v  Found:  Pontophilus norvegicus
## v  Found:  Morone saxatilis
## v  Found:  Buccinum undatum
## v  Found:  Anchoa hepsetus
## v  Found:  Axius serratus
## v  Found:  Pholis gunnellus
## v  Found:  Raja eglanteria
## v  Found:  Decapterus macarellus
## v  Found:  Scomberesox saurus
## v  Found:  Lycenchelys verrilli
## v  Found:  Myoxocephalus aenaeus
## v  Found:  Liparis fabricii
## v  Found:  Monacanthus hispidus
## v  Found:  Spirontocaris liljeborgii
## v  Found:  Sphoeroides maculatus
## v  Found:  Melanostigma atlanticum
## v  Found:  Polymixia lowei
## v  Found:  Xenolepidichthys dalgleishi
## v  Found:  Spirontocaris spinus
## v  Found:  Colus stimpsoni
## v  Found:  Liparis tunicatus
## v  Found:  Phycis chesteri
## v  Found:  Helicolenus dactylopterus
## v  Found:  Ariomma bondi
## v  Found:  Lepophidium profundorum
## v  Found:  Lophius gastrophysus
## v  Found:  Ostrea edulis
## v  Found:  Menticirrhus saxatilis
## v  Found:  Antigonia combatia
## v  Found:  Brosme brosme
## v  Found:  Apeltes quadracus
## v  Found:  Sclerocrangon boreas
## v  Found:  Eumicrotremus spinosus
## v  Found:  Leucoraja garmani
## v  Found:  Ensis directus
## ==  Results  =================
## 
## * Total: 142 
## * Found: 142 
## * Not Found: 0</code></pre>
<pre class="r"><code>info &lt;- matrix(NA)
expand &lt;- matrix(NA)
specific &lt;- matrix(NA, nrow = length(diff_species), ncol=6)


for (i in 1:length(tax)){
info &lt;- tax[[i]][c(&#39;name&#39;,&#39;rank&#39;)]
expand &lt;- info[info$rank == &#39;phylum&#39;| info$rank == &#39;class&#39;| info$rank == &#39;order&#39; | info$rank == &#39;family&#39; | info$rank == &#39;genus&#39; | info$rank == &#39;species&#39;,]
specific[i,] &lt;- as.vector(expand$name)

}
colnames(specific) &lt;- c(&quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;, &quot;Species&quot;)
phylo&lt;-as.data.frame(specific)

merge&lt;-left_join(species,phylo, by=&quot;Species&quot;)
  
landings_tax&lt;-rename(data_regions, Species=SCIENTIFIC_NAME)


#break into groups
east_species&lt;-filter(landings_tax, group==&quot;East&quot;)%&gt;%
  group_by(Year,group,Species)%&gt;%
  summarise(weight=sum(weight))%&gt;%
  mutate(indicator=cur_group_id())%&gt;%
  left_join(phylo)
  
west_species&lt;-filter(landings_tax, group==&quot;West&quot;)%&gt;%
  group_by(Year,group,Species)%&gt;%
  summarise(weight=sum(weight))%&gt;%
  mutate(indicator=cur_group_id())%&gt;%
  left_join(phylo)

pen_species&lt;-filter(landings_tax, group==&quot;Penobscot&quot;)%&gt;%
  group_by(Year,group,Species)%&gt;%
  summarise(weight=sum(weight))%&gt;%
  mutate(indicator=cur_group_id())%&gt;%
  left_join(phylo)

#east
hauls &lt;- unique(east_species$indicator)   
N_hauls &lt;- length(hauls) # number of hauls
N_species &lt;- NULL #N species
sub_species &lt;- NULL # N_species-1
total &lt;- NULL  #xixj
numerator &lt;- NULL #wijxixj
x_y &lt;- matrix(NA, nrow = 6, ncol = 6) 
x &lt;- NULL
y &lt;- NULL
ident &lt;- NULL
weight &lt;- NULL #wij
count &lt;- NULL
total_weight &lt;- NULL
mean_weight &lt;- NULL
weight_var &lt;- NULL
delta &lt;- NULL
delta_star &lt;- NULL
delta_plus &lt;- NULL
delta_var &lt;- NULL
weight_var &lt;- NULL

for (j in 1:N_hauls) {  
diff_hauls &lt;- east_species[which(east_species$indicator == j),] #subset unique hauls/functional groups
N_species[j] &lt;- length(unique(diff_hauls$Species))# count the number of unique species in each haul (denominator)
sub_species[j] &lt;- N_species[j]-1
diff &lt;- unique(as.vector(diff_hauls$Species)) # name of each unique species 
combos &lt;- combn(diff, 2) # create combinations of each species/haul (for weight calc)

phylo &lt;- as.matrix(subset(diff_hauls, select = c(Phylum,Class,Order,Family,Genus, Species))) # extract phylogenetic information only
unique_phylo &lt;- uniquecombs(phylo) # subset by unique species information
unique_phylo &lt;- as.data.frame(unique_phylo)

total &lt;- NULL  # reset the length for each haul because they will be different
weight &lt;- NULL # reset  

for (i in 1:ncol(combos)) { # for each unique combination count the number of each species 
  total[i] &lt;- sum(diff_hauls$Species == combos[1,i]) * sum(diff_hauls$Species == combos[2,i]) #empty vector is always length 210
  #total[i] &lt;- diff_hauls[diff_hauls[,22] == combos[1,i],9] * diff_hauls[diff_hauls[,22] == combos[2,i],9]
  x &lt;- unique_phylo[unique_phylo$Species == combos[1,i],]
  y &lt;- unique_phylo[unique_phylo$Species == combos[2,i],]
  x_y &lt;- rbind(x,y)
  
  for (k in 1:ncol(x_y)){ # for each combination calculate the weight value 
    ident[k] &lt;- identical(as.vector(x_y[1,k]), as.vector(x_y[2,k])) # determine how much of phylogenetic information is the same
    weight[i] &lt;- sum(ident == &quot;FALSE&quot;) # vector of weights
    #mean_weight[i] &lt;- mean(weight) #rep(mean(weight),length(weight))
    numerator[j] &lt;- sum(total*weight) 
    count[j] &lt;- sum(total)
    mean_weight[j] &lt;- mean(weight)
    total_weight[j] &lt;- sum(weight)
    weight_var[j] &lt;- sum((weight- mean(weight))^2) 
  }
  delta &lt;- (2*numerator)/(N_species*sub_species)
  delta_star &lt;- numerator/(count)
  delta_plus &lt;- (2*total_weight)/(N_species*sub_species)
  delta_var &lt;- (2*weight_var)/(N_species*sub_species) #double check that this equation is correct
}
}

years&lt;-2006:2020
delta&lt;-as.data.frame(delta[1:15]) # taxonomic diversity
delta_star&lt;-as.data.frame(delta_star[1:15]) # taxonomic distinctness
delta_plus&lt;-as.data.frame(delta_plus[1:15]) # average taxonomic distinctness
delta_var&lt;-as.data.frame(delta_var[1:15]) # variation in taxonomic distinctness

east_d&lt;-bind_cols(years,delta, delta_star,delta_plus,delta_var)
colnames(east_d)&lt;-c(&quot;year&quot;, &quot;delta&quot;, &quot;delta_star&quot;, &quot;delta_plus&quot;,&quot;delta_var&quot;)

#write.csv(d, here(&quot;Data/tax_metrics.csv&quot;))

east_tax.distinct&lt;-ggplot()+geom_line(data=subset(east_d, year&lt;2020), aes(year, delta_plus))+theme_classic()+ylab(expression(paste(&quot;East\nAverage\nTaxinomic\nDistinctness&quot;)))+theme(plot.margin = margin(10, 10, 10, 35))
                                                                        

                                                                        #west
hauls &lt;- unique(west_species$indicator)   
N_hauls &lt;- length(hauls) # number of hauls
N_species &lt;- NULL #N species
sub_species &lt;- NULL # N_species-1
total &lt;- NULL  #xixj
numerator &lt;- NULL #wijxixj
x_y &lt;- matrix(NA, nrow = 6, ncol = 6) 
x &lt;- NULL
y &lt;- NULL
ident &lt;- NULL
weight &lt;- NULL #wij
count &lt;- NULL
total_weight &lt;- NULL
mean_weight &lt;- NULL
weight_var &lt;- NULL
delta &lt;- NULL
delta_star &lt;- NULL
delta_plus &lt;- NULL
delta_var &lt;- NULL
weight_var &lt;- NULL

for (j in 1:N_hauls) {  
diff_hauls &lt;- west_species[which(west_species$indicator == j),] #subset unique hauls/functional groups
N_species[j] &lt;- length(unique(diff_hauls$Species))# count the number of unique species in each haul (denominator)
sub_species[j] &lt;- N_species[j]-1
diff &lt;- unique(as.vector(diff_hauls$Species)) # name of each unique species 
combos &lt;- combn(diff, 2) # create combinations of each species/haul (for weight calc)

phylo &lt;- as.matrix(subset(diff_hauls, select = c(Phylum,Class,Order,Family,Genus, Species))) # extract phylogenetic information only
unique_phylo &lt;- uniquecombs(phylo) # subset by unique species information
unique_phylo &lt;- as.data.frame(unique_phylo)

total &lt;- NULL  # reset the length for each haul because they will be different
weight &lt;- NULL # reset  

for (i in 1:ncol(combos)) { # for each unique combination count the number of each species 
  total[i] &lt;- sum(diff_hauls$Species == combos[1,i]) * sum(diff_hauls$Species == combos[2,i]) #empty vector is always length 210
  #total[i] &lt;- diff_hauls[diff_hauls[,22] == combos[1,i],9] * diff_hauls[diff_hauls[,22] == combos[2,i],9]
  x &lt;- unique_phylo[unique_phylo$Species == combos[1,i],]
  y &lt;- unique_phylo[unique_phylo$Species == combos[2,i],]
  x_y &lt;- rbind(x,y)
  
  for (k in 1:ncol(x_y)){ # for each combination calculate the weight value 
    ident[k] &lt;- identical(as.vector(x_y[1,k]), as.vector(x_y[2,k])) # determine how much of phylogenetic information is the same
    weight[i] &lt;- sum(ident == &quot;FALSE&quot;) # vector of weights
    #mean_weight[i] &lt;- mean(weight) #rep(mean(weight),length(weight))
    numerator[j] &lt;- sum(total*weight) 
    count[j] &lt;- sum(total)
    mean_weight[j] &lt;- mean(weight)
    total_weight[j] &lt;- sum(weight)
    weight_var[j] &lt;- sum((weight- mean(weight))^2) 
  }
  delta &lt;- (2*numerator)/(N_species*sub_species)
  delta_star &lt;- numerator/(count)
  delta_plus &lt;- (2*total_weight)/(N_species*sub_species)
  delta_var &lt;- (2*weight_var)/(N_species*sub_species) #double check that this equation is correct
}
}

years&lt;-2006:2020
delta&lt;-as.data.frame(delta[1:15]) # taxonomic diversity
delta_star&lt;-as.data.frame(delta_star[1:15]) # taxonomic distinctness
delta_plus&lt;-as.data.frame(delta_plus[1:15]) # average taxonomic distinctness
delta_var&lt;-as.data.frame(delta_var[1:15]) # variation in taxonomic distinctness

west_d&lt;-bind_cols(years,delta, delta_star,delta_plus,delta_var)
colnames(west_d)&lt;-c(&quot;year&quot;, &quot;delta&quot;, &quot;delta_star&quot;, &quot;delta_plus&quot;,&quot;delta_var&quot;)

#write.csv(d, here(&quot;Data/tax_metrics.csv&quot;))

west_tax.distinct&lt;-ggplot()+geom_line(data=subset(west_d, year&lt;2020), aes(year, delta_plus))+theme_classic()+ylab(expression(paste(&quot;West\nAverage\nTaxinomic\nDistinctness&quot;)))+theme(plot.margin = margin(10, 10, 10, 35))
                                                                         
                                                                        
                                                                        #Penobscot
hauls &lt;- unique(pen_species$indicator)   
N_hauls &lt;- length(hauls) # number of hauls
N_species &lt;- NULL #N species
sub_species &lt;- NULL # N_species-1
total &lt;- NULL  #xixj
numerator &lt;- NULL #wijxixj
x_y &lt;- matrix(NA, nrow = 6, ncol = 6) 
x &lt;- NULL
y &lt;- NULL
ident &lt;- NULL
weight &lt;- NULL #wij
count &lt;- NULL
total_weight &lt;- NULL
mean_weight &lt;- NULL
weight_var &lt;- NULL
delta &lt;- NULL
delta_star &lt;- NULL
delta_plus &lt;- NULL
delta_var &lt;- NULL
weight_var &lt;- NULL

for (j in 1:N_hauls) {  
diff_hauls &lt;- pen_species[which(pen_species$indicator == j),] #subset unique hauls/functional groups
N_species[j] &lt;- length(unique(diff_hauls$Species))# count the number of unique species in each haul (denominator)
sub_species[j] &lt;- N_species[j]-1
diff &lt;- unique(as.vector(diff_hauls$Species)) # name of each unique species 
combos &lt;- combn(diff, 2) # create combinations of each species/haul (for weight calc)

phylo &lt;- as.matrix(subset(diff_hauls, select = c(Phylum,Class,Order,Family,Genus, Species))) # extract phylogenetic information only
unique_phylo &lt;- uniquecombs(phylo) # subset by unique species information
unique_phylo &lt;- as.data.frame(unique_phylo)

total &lt;- NULL  # reset the length for each haul because they will be different
weight &lt;- NULL # reset  

for (i in 1:ncol(combos)) { # for each unique combination count the number of each species 
  total[i] &lt;- sum(diff_hauls$Species == combos[1,i]) * sum(diff_hauls$Species == combos[2,i]) #empty vector is always length 210
  #total[i] &lt;- diff_hauls[diff_hauls[,22] == combos[1,i],9] * diff_hauls[diff_hauls[,22] == combos[2,i],9]
  x &lt;- unique_phylo[unique_phylo$Species == combos[1,i],]
  y &lt;- unique_phylo[unique_phylo$Species == combos[2,i],]
  x_y &lt;- rbind(x,y)
  
  for (k in 1:ncol(x_y)){ # for each combination calculate the weight value 
    ident[k] &lt;- identical(as.vector(x_y[1,k]), as.vector(x_y[2,k])) # determine how much of phylogenetic information is the same
    weight[i] &lt;- sum(ident == &quot;FALSE&quot;) # vector of weights
    #mean_weight[i] &lt;- mean(weight) #rep(mean(weight),length(weight))
    numerator[j] &lt;- sum(total*weight) 
    count[j] &lt;- sum(total)
    mean_weight[j] &lt;- mean(weight)
    total_weight[j] &lt;- sum(weight)
    weight_var[j] &lt;- sum((weight- mean(weight))^2) 
  }
  delta &lt;- (2*numerator)/(N_species*sub_species)
  delta_star &lt;- numerator/(count)
  delta_plus &lt;- (2*total_weight)/(N_species*sub_species)
  delta_var &lt;- (2*weight_var)/(N_species*sub_species) #double check that this equation is correct
}
}

years&lt;-2006:2020
delta&lt;-as.data.frame(delta[1:15]) # taxonomic diversity
delta_star&lt;-as.data.frame(delta_star[1:15]) # taxonomic distinctness
delta_plus&lt;-as.data.frame(delta_plus[1:15]) # average taxonomic distinctness
delta_var&lt;-as.data.frame(delta_var[1:15]) # variation in taxonomic distinctness

pen_d&lt;-bind_cols(years,delta, delta_star,delta_plus,delta_var)
colnames(pen_d)&lt;-c(&quot;year&quot;, &quot;delta&quot;, &quot;delta_star&quot;, &quot;delta_plus&quot;,&quot;delta_var&quot;)

#write.csv(d, here(&quot;Data/tax_metrics.csv&quot;))

pen_tax.distinct&lt;-ggplot()+geom_line(data=subset(pen_d, year&lt;2020), aes(year, delta_plus))+theme_classic()+ylab(expression(paste(&quot;Penobscot\nAverage\nTaxinomic\nDistinctness&quot;)))+theme(plot.margin = margin(10, 10, 10, 35))

                                                                        grid.arrange(east_tax.distinct, west_tax.distinct, pen_tax.distinct, nrow=3)                                                                        </code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="east-of-penobscot-bay" class="section level3">
<h3>East of Penobscot Bay</h3>
<pre class="r"><code>#combo plots

#east
r&lt;-ggplot()+geom_line(data=subset(richness, group==&quot;East&quot; &amp; Year&lt;2020),aes(Year, richness))+theme_classic()+ylab(expression(paste(&quot;Species\nRichness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

s&lt;-ggplot()+geom_line(data=subset(shannon, group==&quot;East&quot; &amp; Year&lt;2020),aes(Year, shannon))+theme_classic()+ylab(expression(paste(&quot;Shannon-Weiner\nDiversity&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

e&lt;-ggplot()+geom_line(data=subset(simpson, group==&quot;East&quot; &amp; Year&lt;2020),aes(Year, simpsonE))+theme_classic()+ylab(expression(paste(&quot;Simpson&#39;s\nEvenness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

t&lt;-ggplot()+geom_line(data = subset(east_d, year&lt;2020), aes(year, delta_plus))+theme_classic()+ylab(expression(paste(&quot;Average\nTaxinomic\nDistinctness&quot;)))+theme(plot.margin = margin(10, 10, 10, 25))

grid.arrange(r,s,e,t, nrow=4)</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="west-of-penobscot-bay" class="section level3">
<h3>West of Penobscot Bay</h3>
<pre class="r"><code>#west
r&lt;-ggplot()+geom_line(data=subset(richness, group==&quot;West&quot; &amp; Year&lt;2020),aes(Year, richness))+theme_classic()+ylab(expression(paste(&quot;Species\nRichness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

s&lt;-ggplot()+geom_line(data=subset(shannon, group==&quot;West&quot; &amp; Year&lt;2020),aes(Year, shannon))+theme_classic()+ylab(expression(paste(&quot;Shannon-Weiner\nDiversity&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

e&lt;-ggplot()+geom_line(data=subset(simpson, group==&quot;West&quot; &amp; Year&lt;2020),aes(Year, simpsonE))+theme_classic()+ ylab(expression(paste(&quot;Simpson&#39;s\nEvenness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

t&lt;-ggplot()+geom_line(data = subset(west_d, year&lt;2020), aes(year, delta_plus))+theme_classic()+ylab(expression(paste(&quot;Average\nTaxinomic\nDistinctness&quot;)))+theme(plot.margin = margin(10, 10, 10, 25))

grid.arrange(r,s,e,t, nrow=4)</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
</div>
<div id="penobscot-bay" class="section level3">
<h3>Penobscot Bay</h3>
<pre class="r"><code>#pen bay
r&lt;-ggplot()+geom_line(data=subset(richness, group==&quot;Penobscot&quot; &amp; Year&lt;2020),aes(Year, richness))+theme_classic()+ylab(expression(paste(&quot;Species\nRichness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

s&lt;-ggplot()+geom_line(data=subset(shannon, group==&quot;Penobscot&quot;&amp; Year&lt;2020),aes(Year, shannon))+theme_classic()+ylab(expression(paste(&quot;Shannon-Weiner\nDiversity&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

e&lt;-ggplot()+geom_line(data=subset(simpson, group==&quot;Penobscot&quot;&amp; Year&lt;2020),aes(Year, simpsonE))+theme_classic()+ ylab(expression(paste(&quot;Simpson&#39;s\nEvenness&quot;)))+theme(plot.margin = margin(10, 10, 10, 20))

t&lt;-ggplot()+geom_line(data = subset(pen_d, year&lt;2020), aes(year, delta_plus))+theme_classic()+ylab(expression(paste(&quot;Average\nTaxinomic\nDistinctness&quot;)))+theme(plot.margin = margin(10, 10, 10, 25))

grid.arrange(r,s,e,t, nrow=4)</code></pre>
<p><img src="biodiversity_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
